Sources/BuildSystemIntegration/ExternalBuildSystemAdapter.swift

```diff
@@ -154,11 +153,11 @@ actor ExternalBuildSystemAdapter {
     }
 
     let serverConfig = try BuildServerConfig.load(from: configPath)
-    var serverPath = try AbsolutePath(validating: serverConfig.argv[0], relativeTo: projectRoot)
+    var serverPath = URL(fileURLWithPath: serverConfig.argv[0], relativeTo: projectRoot.ensuringCorrectTrailingSlash)
     var serverArgs = Array(serverConfig.argv[1...])
 
-    if serverPath.suffix == ".py" {
-      serverArgs = [serverPath.pathString] + serverArgs
+    if serverPath.pathExtension == "py" {
+      serverArgs = [try serverPath.filePath] + serverArgs
```

Comment:
Does it make sense to hoist out the `try`?

Reply:
I think the `try` makes sense here because it makes it clear which expression can throw.